extensions/stackdriver/metric/record.cc

```diff
@@ -245,166 +245,309 @@ void addHttpSpecificTags(const ::Wasm::Common::RequestInfo& request_info,
 }
 
 TagKeyValueList getMetricTagMap(
-    TagKeyValueList input_map,
-    std::vector<std::pair<std::string, std::string>> overrides) {
-  if (overrides.size() == 0) {
+    const TagKeyValueList& input_map,
+    const std::vector<std::pair<std::string, std::string>>& tag_overrides) {
+  if (tag_overrides.empty()) {
     return input_map;
   }
 
   TagKeyValueList out;
-  for (auto in : input_map) {
-    auto it = std::find_if(overrides.begin(), overrides.end(),
-                           [&in](const auto& override) {
-                             return override.first == in.first.name();
-                           });
-    if (it != overrides.end()) {
-      auto tag_key = opencensus::tags::TagKey::Register(it->first);
-      out.emplace_back(tag_key, it->second);
+  for (const auto& [tag_key, value_list] : input_map) {
+    auto name = tag_key.name();
+    auto it = std::find_if(
+        tag_overrides.begin(), tag_overrides.end(),
+        [&name](const auto& override) { return override.first == name; });
+    if (it != tag_overrides.end()) {
+      auto key = opencensus::tags::TagKey::Register(it->first);
```

Comment:
This feels awkward.. Can we build a search map for tagKey at registration time?

Reply:
opencensus already does that for us: https://github.com/census-instrumentation/opencensus-cpp/blob/master/opencensus/tags/internal/tag_key.cc#L58-L69

Reply:
It looks very heavy. It has a lock acquisition, and this happen on every call. We should really avoid that.

Reply:
Ok. moved the call to configure().